### PR TITLE
[CLR] Fix Event::awaitCompletion deadlock when worker thread is dead

### PR DESCRIPTION
**Pull request by dixieclick**
`Event::awaitCompletion()` can deadlock if the queue's worker thread is no longer running. Both wait paths — the ActiveWait yield loop and the condition variable wait — spin/block indefinitely because no thread remains to set the event status to `CL_COMPLETE`.

This adds `Os::isThreadAlive(queue->thread())` checks to both wait paths, matching the existing pattern in `HostQueue::terminate()` ([ecea27eb](https://github.com/ROCm/clr/commit/ecea27eb2d935e467ce47b430fa8ef29dfce5f58)).

**Minimum reproducer (Windows):**

```python
import torch
tensor_gpu = torch.randn(1, 1, device="cuda")
# Process hangs on exit — never terminates
```

Adding an explicit `torch.cuda.synchronize()` or `print(tensor_gpu)` (which does an implicit sync via `hipMemcpy`) before exit masks the hang.

`ExitProcess` kills all worker threads before `DLL_PROCESS_DETACH`, then `__hipUnregisterFatBinary` (from a consumer DLL's atexit handler) calls `SyncAllStreams` -> `finish()` -> `awaitCompletion()`, which spins forever.

Tracking:
- https://github.com/pytorch/pytorch/issues/160759
- https://github.com/ROCm/TheRock/issues/999

## Testing

Windows 11, gfx1151, PyTorch ROCm, no Python-level workarounds.

| Test | Result |
|------|--------|
| Unpatched, `torch.randn(1,1,device="cuda")` + exit | Hangs (100%) |
| Patched, `torch.randn(1,1,device="cuda")` + exit | Clean exit (5/5) |
| Patched, heavy GPU workload (Triton JIT compilation + matmul) | Clean exit |

Not tested on Linux (hang does not occur there — `exit()` runs atexit handlers before thread teardown).

## Hang stack

```
ntdll!NtDelayExecution                          <- Os::yield() -> SwitchToThread
KERNELBASE!SwitchToThread
amdhip64_7!...                                  <- Event::awaitCompletion spin
amdhip64_7!_hipUnregisterFatBinary+0x17
ucrtbase!execute_onexit_table                   <- CRT atexit handlers
torch_hip!dllmain_crt_process_detach            <- torch_hip.dll unloading
ntdll!LdrShutdownProcess                        <- ExitProcess (loader lock held)
ntdll!RtlExitUserProcess
kernel32!ExitProcessImplementation
python
```

The deadlock is in `Event::awaitCompletion()` (command.cpp):

```cpp
while (status() > CL_COMPLETE) {   // worker thread is dead — status never
    amd::Os::yield();               // reaches CL_COMPLETE, spins forever
}
```

`HostQueue::finish()` creates a `Marker` command and calls `awaitCompletion()`. The worker thread that would process the Marker and set `status = CL_COMPLETE` was killed by `ExitProcess` before `DLL_PROCESS_DETACH`.

### CODE REVIEW

**Comment by dixieclick** [comment]
@ScottTodd  Looks like the build errors are from ff122e7ed7 (missing YAML config entries for `Unit_hipHostAlloc_AllocateMoreThanTotalSystemMemory` / `Unit_hipHostMalloc_AllocateMoreThanTotalSystemMemory`), not my changes. I'll keep my eye out for any issues after the tests can try again.

**Comment by ScottTodd** [comment]
Yeah... I glanced at the error logs and the workflow status on the base `develop` branch: https://github.com/ROCm/rocm-systems/actions/workflows/therock-ci.yml?query=branch%3Adevelop+event%3Apush. Waiting to see if our build gardener (https://github.com/ROCm/rocm-systems/blob/develop/docs/gardening.md) process notices and fixes that.

Thanks for the PR(s) btw! I've been approving the workflow runs to keep this moving. My hope is that regular contributors/maintainers here in rocm-systems can handle the review and vet your testing plan, but I could also pull these changes down and test them myself as needed.

**Comment by marandje** [comment]
Hi @ScottTodd @dixieclick, my PR had CI jobs passing before this PR (https://github.com/ROCm/rocm-systems/pull/2351) was merged, but now it fails because of missing configuration. I will create a PR to fix that as soon as possible.

**Comment by mangupta** [comment]
> Hi @ScottTodd @dixieclick, my PR had CI jobs passing before this PR (#2351) was merged, but now it fails because of missing configuration. I will create a PR to fix that as soon as possible.

@marandje has fixed the issue via #3687.

**Comment by dixieclick** [comment]
@ScottTodd Looks like another test should work now?  Hopefully it passes.  Windows should work without issue.  I don't have a Linux environment set up at the moment for ROCm so can't easily test that.

**Comment by dixieclick** [comment]
The merged PR #3790 fixes this at higher level.  Closing this.

**Comment by ScottTodd** [comment]
Thanks again for proposing these fixes, even though that other PR went in instead. I'm happy to see this get resolved now :)
